torchaudio/functional.py

```diff
@@ -2082,7 +2083,7 @@ def _measure(
         if boot_count >= 0 \
         else measure_smooth_time_mult
 
-    _d = complex_norm(_dftBuf[spectrum_start:spectrum_end])
```

Comment:
I'd keep the change norm/abs separate. In a prior attempt, there was a performance regression, #747.

Reply:
@vincentqb Can you elaborate why the regression in `torch.norm`, which is not used in this PR, is the reason to discourage the use of `torch.abs`? Are they using the same implementation under the hood?

Reply:
I'll let @anjali411 @mruberry comment on how `torch.abs` and `torch.norm` are related if they are. If they are we would end up with a performance regression again. We could add some performance tests/checks manually or automatically to catch this. However, my suggestion is simply to decouple the changes about `torch.*fft*` from those about the use of `.abs()` so that, in the event we do get a regression, we can easily track it and revert it.

Reply:
Datapoint: torch.abs and torch.norm have separate implementations.

Reply:
I think it should be perfectly fine to use `torch.abs()` here. `complex_norm` is not even using `torch.norm`. In fact it's using three kernels (pow, sum, pow), so `torch.abs()` should be a strict improvement here.

Reply:
> Datapoint: torch.abs and torch.norm have separate implementations.

Good to know :)

> I think it should be perfectly fine to use `torch.abs()` here. `complex_norm` is not even using `torch.norm` https://github.com/pytorch/audio/blob/master/torchaudio/functional.py#L424. In fact, it's using three kernels (pow, sum, pow), so `torch.abs()` should be a strict improvement here.

We had `complex_norm` using `torch.norm`, but this led to a speed regression so we revert to the current implementation with three kernels. Because of this, I want to avoid a performance regression again. I'm glad to know this would launch only one kernel: has the performance difference been tested and compared in this case?

In any case, let's just separate concerns and move this to a separate pull request so we don't block the rest on this :)

Reply:
```
x=torch.randn(10, 20, 2)
In [7]: def fn(x):
   ...:     t0=time.time()
   ...:     o=x.pow(2.0).sum(-1).pow(0.5)
   ...:     t1 = time.time()
   ...:     print(t1-t0)
   ...:

In [8]: fn(x)
0.00037217140197753906

In [10]: y=torch.view_as_complex(x).contiguous()

In [11]: fn(y)
0.0001480579376220703
```

There's a significant performance gain (as expected) so I think we should switch to `torch.abs`.

Reply:
Thanks @anjali411 !

In addition to that I did benchmark for the exact code path that @vincentqb suggested and observed that `abs` is 10x faster.

| Method                          | Output                                   |
|---------------------------------|------------------------------------------|
| `t.abs()`                       | 100 loops, best of 5: 1.32 msec per loop |
| `complex_norm(view_as_real(t))` | 100 loops, best of 5: 14.1 msec per loop |

<details><summary>code</summary>

```bash
OMP_NUM_THREADS=1 numactl --membind 0 --cpubind 0 python -m timeit -n 100 -r 5 -s """
import torch;
import torch.fft;
torch.manual_seed(0);
t = torch.fft.rfft(torch.randn(1, 32*44100));
""" """
t.abs();
"""

OMP_NUM_THREADS=1 numactl --membind 0 --cpubind 0 python -m timeit -n 100 -r 5 -s """
import torch;
import torch.fft;
import torchaudio.functional;
torch.manual_seed(0);
t = torch.fft.rfft(torch.randn(1, 32*44100));
""" """
torchaudio.functional.complex_norm(torch.view_as_real(t));
"""
```

</details>

Reply:
Thanks for checking!